sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -59,6 +60,7 @@ case class PartitionedFile(
     filePath: SparkPath,
     start: Long,
     length: Long,
+    fileStatus: FileStatus,
```

Comment:
Similarly, due to the addition of `fileStatus`, the constructor of `PartitionedFile` can also be further simplified, right?

Reply:
In addition, since `fileStatus` will hold more state and also participate in serialization, will this lead to additional memory overhead and serialization pressure?

Reply:
The fileStatus should occupy more memory, but I haven't received OOM issues during the rollout of this change to the online cluster.

Reply:
@cloud-fan Are there also risks of breaking internal APIs with modifications similar to those made here and in `FileFormat.createMetadataInternalRow`?